### PR TITLE
task4591 edit profile password

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4591

### CODE REVIEW

**Comment by laam55** [comment]
ready

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -41,9 +45,91 @@ public function detail($id)
...
+    public function editProfile(EditProfileRequest $request)
+    {
+        $input = $request->all();
+        $currentUser = Auth::user();
+
+        if (!empty($input)) {
+
+            if ($this->userRepository->update($input, $currentUser->id)) {
+                return redirect()->route('home.user.setting');
+            }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/UserController.php]: convention

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -41,9 +45,91 @@ public function detail($id)
...
+            $fileName2 =  md5($currentUser->id . '_' . $fileName1) . '.' . $input['image']->getClientOriginalExtension();
+            $input['image_path'] = $fileName2;
+
+            if ($this->userRepository->update($input, $currentUser->id)) {
+                // move file to uploads folder
+                $input['image']->move(Config::get('constants.IMAGE_UPLOAD_PATH'), $fileName2);
+                Session::flash('success_alert', __('alert.success.update'));
+
+                return redirect()->route('home.user.setting');
+            }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/UserController.php]: convention

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -41,9 +45,91 @@ public function detail($id)
...
+                $input['password'] = Hash::make($input['password']);
+
+                if ($this->userRepository->update($input, $currentUser->id)) {
+                    //set new password to Auth
+                    $currentUser['password'] = $input['password'];
+                    Auth::setUser($currentUser);
+                    Session::flash('success_alert', __('alert.success.update'));
+
+                    return redirect()->route('home.user.password');
+                }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/UserController.php]: convention

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -41,9 +45,91 @@ public function detail($id)
...
+                    return redirect()->route('home.user.password')->withErrors(['errorUpdate' => __('alert.error.update')]);
+                }
+            }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/UserController.php]: convention

File: app/Http/Requests/EditPasswordRequest.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public function rules()
+    {
+        return [
+            'oldPassword' => 'required|between:6,20',
+            'password' => 'required|between:6,20|confirmed',
+        ];
+    }
+
+    public function messages($value='')
```
Comment (BunnyPi04) [inline app/Http/Requests/EditPasswordRequest.php]: convention

**Pull request merged**